### DISCUSSION TITLE
Add a timer

### DISCUSSION THREAD

**Discussion by Soif2Sang** [flet-dev/flet#750] [Q&A] [answered]
Hello, I'm looking for a way to add a decreasing timer without using any time.sleep(),
Do you have any idea how to achieve this ?
Thank you !

**Comment by Francois-Belanger** [comment]
I use something like this.  I cannot share the code I did, but it's close.

https://stackoverflow.com/questions/45419723/python-timer-with-asyncio-coroutine

**Reply by Francois-Belanger** [reply]
Maybe: https://pypi.org/project/atimer/

**Reply by Soif2Sang** [reply]
Thank you for the reply, it might be what i'm looking for. :)

**Comment by FeodorFitsner** [comment, marked as answer]
See the last example here: https://flet.dev/docs/guides/python/user-controls It uses `time.sleep()` though, but why would you like to avoid it?

**Reply by Soif2Sang** [reply]
It was exactly what I'm looking for, thank you very much ! 
I didn't want to use sleep time because it makes the interface non responsive during the sleeping time, but using threads solve this issue.
Again, thanks a lot :)
